### PR TITLE
Feature/improve block prefetching in collect

### PR DESCRIPTION
**Pull request by mchacki**
Minor improvement in Collect nodes.
Nothing visible to the outside, it just might improve the speed of some queries slightly.

I have not written changeling, if we decide to use this imrpvoment I will add CHANGELOG to remove the DO_NOT_MERGE flag.

Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8692/

### CODE REVIEW

**Comment by mchacki** [comment]
New Jenkins including the regression test:

http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8710/

File: CHANGELOG [hunk 1]
```diff
@@ -1,13 +1,21 @@
 devel
 -----
 
+* In some cases with a COLLECT LIMIT situation on a small limit the collect
+  does more calls to upstream then without a limit to provide the same
```
Comment (jsteemann) [inline CHANGELOG]:
```suggestion
  does more calls to upstream than without a limit to provide the same
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+  result. We improved this situation and made sure that LIMIT does
+  not cause the COLLECT to fetch too few input. There is a chance
```
Comment (jsteemann) [inline CHANGELOG]:
```suggestion
  not cause the COLLECT to fetch too few input rows. There is a chance
```

File: CHANGELOG [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
+  modification, most queries will benefit however.  
+
 * Use OpenSSL's EVP interface for SHA256 instead of the deprecated low-level
   message digest APIs.
 
 * Prevent startup error "specifiy either '--server.jwt-secret-keyfile' or 
   '--server.jwt-secret-folder' but not both." from occurring when specifying the
   JWT secret keyfile option.
 
+
```
Comment (jsteemann) [inline CHANGELOG]:
```suggestion
```

File: CHANGELOG (hunk 2 above)
Comment (jsteemann) [reply CHANGELOG]:
```suggestion
  not cause the COLLECT to fetch too few input rows. There is a chance
```

**Comment by jsteemann** [comment]
@mchacki : thanks!

**Pull request merged**